web-services-app/src/main/java/edu/unc/lib/boxc/web/services/processing/IiifV3ManifestService.java

```diff
@@ -167,21 +219,65 @@ private Canvas constructCanvasSection(ContentObjectRecord contentObj) {
         imageContent.setServices(imageService);
 
         // Set the dimensions of this item on appropriate elements
-        assignDimensions(contentObj, canvas, imageContent);
+        assignImageDimensions(contentObj, canvas, imageContent);
+    }
 
-        return canvas;
+    private void assignImageDimensions(ContentObjectRecord contentObj, Canvas canvas, ImageContent imageContent) {
+        var dimensions = getDimensions(contentObj);
+        if (dimensions != null) {
+            var width = dimensions.get(WIDTH);
+            var height = dimensions.get(HEIGHT);
+            canvas.setWidthHeight(width, height); // Dimensions for the canvas
+            imageContent.setWidthHeight(width, height); // Dimensions for the actual image
+        }
     }
 
-    private void assignDimensions(ContentObjectRecord contentObj, Canvas canvas, ImageContent imageContent) {
-        Datastream fileDs = contentObj.getDatastreamObject(DatastreamType.ORIGINAL_FILE.getId());
+    private Map<String, Integer> getDimensions(ContentObjectRecord contentObj) {
+        var fileDs = getFileDatastream(contentObj);
         String extent = fileDs.getExtent();
-        if (extent != null && !extent.equals("")) {
+        if (extent != null && !extent.isEmpty()) {
             String[] imgDimensions = extent.split("x");
-            var width = Integer.parseInt(imgDimensions[1]); // in the datastream, the width is second
-            var height = Integer.parseInt(imgDimensions[0]);
-            canvas.setWidthHeight(width, height); // Dimensions for the canvas
-            imageContent.setWidthHeight(width, height); // Dimensions for the actual image
+            var imgDimensionsCount = Arrays.stream(imgDimensions).count();
+            if (imgDimensionsCount < 2) {
+                return null;
+            }
+            return extractDimensions(imgDimensions, imgDimensionsCount);
+        }
+        return null;
+    }
+
+    private static HashMap<String, Integer> extractDimensions(String[] imgDimensions, long imgDimensionsCount) {
```

Comment:
```suggestion
    private static Map<String, Integer> extractDimensions(String[] imgDimensions) {
```

Changed HashMap to Map, and we don't really need to pass in the count since we can do imgDimensions.length